backend/src/v2/compiler/argocompiler/container.go

```diff
@@ -121,6 +122,24 @@ func (c *workflowCompiler) addContainerDriverTemplate() string {
 			Resources: driverResources,
 		},
 	}
+	// Add retry policy
+	if retryPolicy != nil {
+		maxRetryCount := intstr.FromInt(int(retryPolicy.GetMaxRetryCount()))
+		// During the converstion from float64 to int, the fraction is discarded (truncated towards 0).
+		// See https://go.dev/ref/spec#Conversions
+		backOffFactor := intstr.FromInt(int(retryPolicy.GetBackoffFactor()))
```

Comment:
I am probably missing something. Why do we need to use `int()` and  `intstr.FromInt()` for these parameters (`int32` and `double` with default values of `0`)?